kilkari/src/main/java/org/motechproject/nms/kilkari/service/impl/CsrServiceImpl.java

```diff
@@ -156,31 +156,62 @@ private boolean isMsisdnInvalid(CallSummaryRecord record) {
     }
 
 
+    private boolean isFirstMessageOfWeek(String weekId) {
+        return weekId.endsWith("1");
+    }
+
+
     private void rescheduleCall(Subscription subscription, CallSummaryRecord record, CallRetry callRetry) {
 
         Long msisdn = subscription.getSubscriber().getCallingNumber();
 
-        if (callRetry == null) {
-
-            // This message was never retried before, so this is a first retry
-            LOGGER.debug(String.format("Creating retry for subscription: %s", subscription.getSubscriptionId()));
-            DayOfTheWeek nextDay = DayOfTheWeek.fromInt(subscription.getStartDate().dayOfWeek().get()).nextDay();
-            CallRetry newCallRetry = new CallRetry(
-                    subscription.getSubscriptionId(),
-                    msisdn,
-                    nextDay,
-                    CallStage.RETRY_1,
-                    record.getContentFileName(),
-                    record.getWeekId(),
-                    getLanguageCode(subscription),
-                    getCircleName(subscription),
-                    subscription.getOrigin()
-            );
-            callRetryDataService.create(newCallRetry);
+        if (callRetry == null || !callRetry.getWeekId().equals(record.getWeekId())) {
+
+            // This message was never retried for this week before, so this is a first retry
+            LOGGER.debug(String.format("Creating retry for subscription %s and weekId %s",
+                    subscription.getSubscriptionId(), record.getWeekId()));
+
+            // Looking at the weekId, we can figure out if this was the first or second message for that week
+            // and calculate which day of the week we should reschedule the call for
+            DayOfTheWeek nextDay;
+            if (isFirstMessageOfWeek(record.getWeekId())) {
+                nextDay = subscription.getFirstMessageDayOfWeek().nextDay();
```

Comment:
The nextDay is irrelevant now right? The only reason we care about first vs second message/week is to set/update the retry call stage?